server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -156,8 +159,9 @@ private void removeCandidatesInUse(GarbageCollectionEnvironment gce,
 
         dir = makeRelative(dir, 2);
 
-        if (candidateMap.remove(dir) != null) {
-          log.debug("Candidate was still in use: {}", dir);
+        GcCandidate gcTemp = candidateMap.remove(dir);
+        if (gcTemp != null) {
+          log.debug("Directory Candidate was still in use by dir ref: {}", dir);
```

Comment:
Not sure this is the best wording for the comment, but would be useful to add something indicating that not adding to inUseCandidates is intentional

```suggestion
          log.debug("Directory Candidate was still in use by dir ref: {}", dir);
          // intentionally not adding directory candidates to inUseCandidates because dir candidates are only added once if deleted because in use nothing will add them again.
```

Reply:
Comments have been added regarding the intentional ignoring of the InUse Dir candidates.